app/src/main/java/com/example/trainingintive/di/RoomModule.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.trainingintive.di
+
+import android.content.Context
+import androidx.room.Room
+import com.example.trainingintive.repository.local.ApplicationDatabase
+import com.example.trainingintive.repository.local.DogImageDao
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+
+private const val DATABASE_NAME = "application_database"
+
+@Module
+class RoomModule {
+
+    @Provides
+    @Singleton
+    fun provideDatabase(context: Context): ApplicationDatabase {
+        return Room.databaseBuilder(
+            context.applicationContext,
+            ApplicationDatabase::class.java,
+            DATABASE_NAME
+        ).build()
+    }
+
+    @Provides
+    @Singleton
+    fun provideDogImageUrlDao(applicationDatabase: ApplicationDatabase): DogImageDao = applicationDatabase.dogImageUrlDao()
```

Comment:
It would be more readable if you wrap it to new line, like this:
```
fun provideDogImageUrlDao(
    applicationDatabase: ApplicationDatabase
) = applicationDatabase.dogImageUrlDao()
```
or
```
fun provideDogImageUrlDao(applicationDatabase: ApplicationDatabase) =
 applicationDatabase.dogImageUrlDao()
```